packages/plugins/experimental/plugin-outliner/src/types/types.ts

```diff
@@ -1,28 +1,49 @@
 //
-// Copyright 2023 DXOS.org
+// Copyright 2024 DXOS.org
 //
 
-import { S } from '@dxos/echo-schema';
-
-import { TreeType } from './tree';
-import { OUTLINER_PLUGIN } from '../meta';
-
-export namespace OutlinerAction {
-  const OUTLINER_ACTION = `${OUTLINER_PLUGIN}/action`;
-
-  export class Create extends S.TaggedClass<Create>()(`${OUTLINER_ACTION}/create`, {
-    input: S.Struct({
-      name: S.optional(S.String),
-    }),
-    output: S.Struct({
-      object: TreeType,
-    }),
-  }) {}
-
-  export class ToggleCheckbox extends S.TaggedClass<ToggleCheckbox>()(`${OUTLINER_ACTION}/toggle-checkbox`, {
-    input: S.Struct({
-      object: TreeType,
-    }),
-    output: S.Void,
-  }) {}
-}
+import { Ref, S, TypedObject, type Ref$ } from '@dxos/echo-schema';
+
+// TODO(burdon): Reconcile with Task. Convert from TypedObject to Schema.
+// TODO(burdon): Change -Type suffix to -Schema?
+
+export class TreeNodeType extends TypedObject({ typename: 'dxos.org/type/TreeNode', version: '0.1.0' })({
+  // TODO(burdon): Refs vs. objects?
+  children: S.suspend((): S.mutable<S.Array$<Ref$<TreeNodeType>>> => S.mutable(S.Array(Ref(TreeNodeType)))),
+
+  text: S.String,
+
+  // TODO(burdon): Support mixin.
+  done: S.optional(S.Boolean),
+}) {}
+
+export class TreeType extends TypedObject({ typename: 'dxos.org/type/Tree', version: '0.1.0' })({
+  name: S.optional(S.String),
+  root: Ref(TreeNodeType), // TOOD(burdon): Remove Ref.
+}) {}
+
+// export const JournalEntryType = S.Struct({
+//   date: S.Date,
+//   root: TreeNodeType,
+// }).pipe(EchoObject('dxos.org/type/JournalEntry', '0.1.0'));
+
+// export type JournalEntryType = S.Schema<typeof JournalEntryType>;
+
+// export const JournalType = S.Struct({
+//   name: S.optional(S.String),
+//   entries: S.mutable(S.Array(JournalEntryType)),
+// }).pipe(EchoObject('dxos.org/type/Journal', '0.1.0'));
+
+// export type JournalType = S.Schema<typeof JournalType>;
+
+export class JournalEntryType extends TypedObject({ typename: 'dxos.org/type/JournalEntry', version: '0.1.0' })({
+  date: S.Date,
+  root: TreeNodeType,
+}) {}
+
+export class JournalType extends TypedObject({ typename: 'dxos.org/type/JournalType', version: '0.1.0' })({
+  name: S.optional(S.String),
+
+  // TODO(burdon): Refs vs. objects?
+  entries: S.mutable(S.Array(JournalEntryType)),
```

Comment:
Can't have an echo object be embedded in another typed object (we can discuss how to allow that separatelly). This either needs to be a ref or JournalEntryType to not be an echo object.

Reply:
the storybook throws: not an ECHO object error.
http://localhost:9009/?path=/story/plugins-plugin-outliner-outliner--large